core/modules/field_group/field_group.field_ui.inc

```diff
@@ -0,0 +1,830 @@
+      $params->parents[$child] = $name;
+    }
+  }
+
+  return $params;
+}
+
+/**
+ * Function to alter the fields overview and display overview screen.
+ * Does not implement a hook.
+ */
+function field_group_field_ui_overview_form_alter(&$form, &$form_state, $display_overview = FALSE) {
+  // Only start altering the form if we need to.
+  if ($display_overview && empty($form['#fields']) && empty($form['#extra'])) {
+    return;
+  }
+
+  $params = field_group_field_ui_form_params($form, $display_overview);
+
+  // Add some things to be able to preserve synced usage of field_ui.
+  if (!$display_overview) {
+    // This key is used to store the current updated field.
+    $form_state += array(
+      'formatter_settings_edit' => NULL,
+    );
+    // Add AJAX wrapper.
+    $form['fields']['#prefix'] = '<div id="field-display-overview-wrapper">';
+    $form['fields']['#suffix'] = '</div>';
+  }
+  $form['#groups'] = array_keys($params->groups);
+
+  $table = &$form['fields'];
+
+  // Add a region for 'add_new' rows, but only when fields are
+  // available and thus regions.
+  if (isset($table['#regions'])) {
+    $table['#regions'] += array(
+      'add_new' => array('title' => '&nbsp;'),
+    );
+  }
+
+  // Extend available parenting options.
+  foreach ($params->groups as $name => $group) {
+    $table['#parent_options'][$name] = $group->label;
+  }
+  $table['#parent_options']['_add_new_group'] = t('Add new group');
+
+  // Update existing rows accordingly to the parents.
+  foreach (element_children($table) as $name) {
+    $table[$name]['parent_wrapper']['parent']['#options'] = $table['#parent_options'];
+    // Inherit the value of the parent when default value is empty.
+    if (empty($table[$name]['parent_wrapper']['parent']['#default_value'])) {
+      $default = '';
+      if (isset($params->parents[$name])) {
+        $default = $params->parents[$name];
+      }
+      $table[$name]['parent_wrapper']['parent']['#default_value'] = $default;
+    }
+  }
+
+  $formatter_options = field_group_field_formatter_options($display_overview ? 'display' : 'form');
+
+  $refresh_rows = NULL;
+  if (isset($form_state['values']['refresh_rows'])) {
+    $refresh_rows = $form_state['values']['refresh_rows'];
+  }
+  elseif (isset($form_state['input']['refresh_rows'])) {
+    $refresh_rows = $form_state['input']['refresh_rows'];
+  }
+
+  // Create the group rows and check actions.
+  foreach (array_keys($params->groups) as $name) {
+
+    // Play around with form_state so we only need to hold things
+    // between requests, until the save button was hit.
+    if (isset($form_state['field_group'][$name])) {
+      $group = & $form_state['field_group'][$name];
+    }
+    else {
+      $group = & $params->groups[$name];
+    }
+
+    // Check the currently selected formatter, and merge persisted values for
+    // formatter settings for the group.
+    // This needs to be done first, so all fields are updated before creating
+    // form elements.
```

Comment:
```suggestion
    // Check the currently selected formatter, and merge persisted values for
    // formatter settings for the group. This needs to be done first, so that
    // all fields are updated before creating form elements.
```